design-proposals/eim-cli.md

```diff
@@ -0,0 +1,254 @@
+
+#### Provision Host
+
+This option of the `create` command will support provisioning of any onboarded host using the `--provision` option followed by arguments needed to successfully provision a registered host.
+
+*Required info: `OSProfile`/`Site`/`Secure`/`RemoteUser`/`Metadata`*
+
+- Provision Host (ie. `emfctl create host myhost --provision <arguments>`)
+
+> NOTE: Need to establish what the default behaviour should be for `create host myhost` without argument (in UI it is equivalent of `--provision <args>` + `--auto-onboarding`)
+
+#### Edit Host
+
+Registered/Onboarded/Provisioned hosts can be edited
+
+*Required info: `Name`/`Site`/`Region`/`Metadata`*
+
+- Edit host (ie `emfctl edit host myhost --edit-config <arguments>`)
+
+
+#### Deauthorize and/or delete Host
+
+- Deauthorise Host (ie. `emfctl delete host myhost` `--deauthorize-only`)
+> TODO: Is this correct default behaviour?
+- Deauthorise and Delete Host (ie. `emfctl delete host myhost`)
+- Delete Host (ie. `emfctl delete host myhost --force-delete`)
+
+#### Other
+
+> **NOTE/QUESTION**: Viewing metrics, scheduling maintenance are not considered to be supported in the initial phase of the EIM related CLI commands - should they be?
+
+### Ability to create & manage Locations/regions/sub-regions/sites
+
+#### Local options
+
+The local options associated with the `location`/`region`/`sub-region`/`site` object/noun would help with filtering and providing arguments to the commands:
+
+- `--region-type` - to be used with `create region/subregion`
+- `--region` `<arg>` - region name/names to be used with sub-region
+- `--latitude` - latitude for site
+- `--longititude` - longtitude for site
+
+> NOTE: TODO Do we need to support advanced region settings in this release?
+
+#### List locations/regions/sub-regions/sites
+
+> NOTE: TODO Not yet sure what is actually useful for the user to list vs how it's going to be used to edit and manage the regions. Is there a point in having different commands to list different levels of location where individually they do not contain all that much info if advanced settings are not enabled?
+
+List locations will display the hierarchical tree of the location created
+
+- List locations (ie. `emfctl list location`)
+- List all regions only (ie. `emfctl list region`)
+- List all sub-regions only (ie. `emfctl list sub-region`)
+- List all sites only (ie. `emfctl list site`)
+
+#### Get regions/sub-regions/sites
+
+Get region details including sub regions and sites below.
+
+- Get region including sub regions and sites below. (ie. `emfctl get region myregion`)
+- Get sub-region including sub regions and sites below. (`emfctl get sub-region mysubregion --region myregion mysubregion mysubregionssubregion`)
+- Get site details including the regions/subregions above (`emfctl get site mysite --region myregion mysubregion`)
```

Comment:
not sure I understand here.

I would rather consider the implemenation generic: ad hoc parameter to return hierarchy, parameter fo the max depth. This can span region/site and the return entities might include hosts as leaf